pyfr/readers/cgns.py

```diff
@@ -0,0 +1,423 @@
+        # cg_open
+        lib.cg_open.argtypes = [c_char_p, c_int, POINTER(c_int)]
+        lib.cg_open.errcheck = self._errcheck
+
+        # cg_close
+        lib.cg_close.argtypes = [c_int]
+        lib.cg_close.errcheck = self._errcheck
+
+        # cg_nbases
+        lib.cg_nbases.argtypes = [c_int, POINTER(c_int)]
+        lib.cg_nbases.errcheck = self._errcheck
+
+        # cg_base_read
+        lib.cg_base_read.argtypes = [c_int, c_int, c_char_p, POINTER(c_int),
+                                     POINTER(c_int)]
+        lib.cg_base_read.errcheck = self._errcheck
+
+        # cg_nzones
+        lib.cg_nzones.argtypes = [c_int, c_int, POINTER(c_int)]
+        lib.cg_nzones.errcheck = self._errcheck
+
+        # cg_zone_read
+        lib.cg_zone_read.argtypes = [c_int, c_int, c_int, c_char_p,
+                                     POINTER(c_int)]
+        lib.cg_zone_read.errcheck = self._errcheck
+
+        # cg_zone_type
+        lib.cg_zone_type.argtypes = [c_int, c_int, c_int, POINTER(c_int)]
+        lib.cg_zone_type.errcheck = self._errcheck
+
+        # cg_coord_read
+        lib.cg_coord_read.argtypes = [
+            c_int, c_int, c_int, c_char_p, c_int, POINTER(c_int),
+            POINTER(c_int), c_void_p
+        ]
+        lib.cg_coord_read.errcheck = self._errcheck
+
+        # cg_nbocos
+        lib.cg_nbocos.argtypes = [c_int, c_int, c_int, POINTER(c_int)]
+        lib.cg_nbocos.errcheck = self._errcheck
+
+        # cg_boco_info
+        lib.cg_boco_info.argtypes = [
+            c_int, c_int, c_int, c_int, c_char_p, POINTER(c_int),
+            POINTER(c_int), POINTER(c_int), POINTER(c_int),
+            POINTER(c_int), POINTER(c_int), POINTER(c_int)
+        ]
+        lib.cg_boco_info.errcheck = self._errcheck
+
+        # cg_boco_read
+        lib.cg_boco_read.argtypes = [c_int, c_int, c_int, c_int,
+                                     POINTER(c_int), c_void_p]
+        lib.cg_boco_read.errcheck = self._errcheck
+
+        # cg_nsections
+        lib.cg_nsections.argtypes = [c_int, c_int, c_int, POINTER(c_int)]
+        lib.cg_nsections.errcheck = self._errcheck
+
+        # cg_section_read
+        lib.cg_section_read.argtypes = [
+            c_int, c_int, c_int, c_int, c_char_p, POINTER(c_int),
+            POINTER(c_int), POINTER(c_int), POINTER(c_int), POINTER(c_int)
+        ]
+        lib.cg_section_read.errcheck = self._errcheck
+
+        # cg_ElementDataSize
+        lib.cg_ElementDataSize.argtypes = [c_int, c_int, c_int, c_int,
+                                           POINTER(c_int)]
+        lib.cg_ElementDataSize.errcheck = self._errcheck
+
+        # cg_elements_read
+        lib.cg_elements_read.argtypes = [c_int, c_int, c_int, c_int,
+                                         c_void_p, c_void_p]
+        lib.cg_elements_read.errcheck = self._errcheck
+
+    def _hdf5_load(self):
+        try:
+            load_library('hdf5', mode=RTLD_GLOBAL)
+        except OSError:
+            import pkgconfig
```

Comment:
Wrap this in a try except as otherwise we could give some misleading exceptions.